src/plugins/intel_gpu/src/plugin/ops/split.cpp

```diff
@@ -17,40 +17,43 @@ static void CreateCommonSplitOp(Program& p, const std::shared_ptr<ngraph::Node>&
     auto inputPrimitives = p.GetInputPrimitiveIDs(op);
     std::string layerName = layer_type_name_ID(op);
 
-    auto input_pshape = op->get_input_partial_shape(0);
-    OPENVINO_ASSERT(input_pshape.is_static(),
-                    "Dynamic shapes are not supported yet for v1::Split and v1::VariadicSplit operations");
-
-    auto input_shape = input_pshape.to_shape();
-    InferenceEngine::SizeVector start_offset(input_shape.size());
+    auto inPartialShape = op->get_input_partial_shape(0);
+    InferenceEngine::SizeVector startOffset(inPartialShape.size());
 
     bool is_single_out_split = op->get_output_size() == 1;
 
     for (size_t i = 0; i < op->get_output_size(); i++) {
         std::string outLayerName = layerName + (is_single_out_split ? "" : ".out" + std::to_string(i));
-        const auto outLayerDims = op->get_output_shape(i);
-        NGRAPH_SUPPRESS_DEPRECATED_START
-        if (outLayerDims.size() != start_offset.size()) {
-            IE_THROW() << "Invalid dimesions in split layer: " << op->get_friendly_name()
-                               << " output: " <<  op->get_output_tensor_name(i);
-        }
-        for (size_t i = 0; i < input_shape.size(); i++) {
-            if ((outLayerDims[i] + start_offset[i]) > input_shape[i]) {
+        const auto outPatialShape = op->get_output_partial_shape(i);
+        if (outPatialShape.is_static()) {
+            NGRAPH_SUPPRESS_DEPRECATED_START
+            if (outPatialShape.size() != startOffset.size()) {
                 IE_THROW() << "Invalid dimesions in split layer: " << op->get_friendly_name()
-                                   << " output: " <<  op->get_output_tensor_name(i);
+                                << " output: " <<  op->get_output_tensor_name(i);
+            }
+            for (size_t i = 0; i < inPartialShape.size(); i++) {
+                if ((outPatialShape[i].get_length() + static_cast<ov::Dimension::value_type>(startOffset[i])) > inPartialShape[i].get_length()) {
+                    IE_THROW() << "Invalid dimesions in split layer: " << op->get_friendly_name()
+                                    << " output: " <<  op->get_output_tensor_name(i);
+                }
             }
+            NGRAPH_SUPPRESS_DEPRECATED_END
         }
-        NGRAPH_SUPPRESS_DEPRECATED_END
 
-        auto outTensor = tensor_from_dims(outLayerDims, 1);
-        auto offsetTensor = tensor_from_dims(start_offset, 0);
-
-        auto cropPrim = cldnn::crop(outLayerName, inputPrimitives[0], outTensor, offsetTensor);
+        auto outTensor = outPatialShape.is_static()? tensor_from_dims(outPatialShape.to_shape(), 1) : cldnn::tensor(1);
+        auto offsetTensor = tensor_from_dims(startOffset, 0);
+        auto cropPrim = cldnn::crop(outLayerName, inputPrimitives, outTensor, offsetTensor, i);
```

Comment:
Also I think we can pass inputPrimitives[0] for static shape and pass inputPrimitives for dynamic shape. 
Because static shape does not need runtime offset while the internal impl is using the # of dependencies to check whether it should calculate runtime offset or not.

Reply:
I think # of dependencies (or # of inputPrimitives) is defined by the original ngraph operation. Do you mean outTensor and offsetTensor doesn't need when input is dynamic shape ?

Reply:
@yeonbok , I upload fixed version. could you please confirm is it what you say?

Reply:
Done